python/ray/serve/tests/test_target_capacity.py

```diff
@@ -309,5 +322,255 @@ def test_autoscaling_scale_to_zero(
     )
 
 
+@serve.deployment(
+    ray_actor_options={"num_cpus": 0},
+    max_concurrent_queries=2,
+    graceful_shutdown_timeout_s=0,
+)
+class HangDeployment:
+    async def __call__(self, *args):
+        await asyncio.sleep(10000)
```

Comment:
Non-blocker: Might be good to use `SignalActor` here. But also feel it's pretty much equivalent XD

Reply:
I'd prefer to keep it as-is. There's less boilerplate code, and it's clear that any requests sent to the deployment aren't meant to finish.